### ISSUE TITLE
how to return ids of affected rows on delete

### ISSUE THREAD

**Issue by salismt** [jdbi/jdbi#1620]
I have statement to delete all transactions where it belongs to a card

```
@SqlUpdate("DELETE FROM transaction WHERE card_id = :card_id AND made_on IS NULL")
@GetGeneratedKeys("id")
List<String> purgePendingTransaction(@Bind("card_id") String cardId);
```

where the Id is string, but I cannot do it with list, I can only get the last affected id
I want to retrieve all the affected id.

How to do it?

I am using jdbi 3.10.1

**Comment by leaumar** [comment]
AFAIK this depends entirely on your database and driver, not something Jdbi can promise to improve for you. You could instead start a transaction, query the target ids, and run a delete statement with those ids.

**Comment by stevenschlansker** [comment]
Yes, could you please provide additional details: what database, what database driver, and versions?

We do have a test that this feature works in principle:
https://github.com/jdbi/jdbi/blob/master/sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestGetGeneratedKeysPostgres.java#L106-L114

Here, you can see us retrieving a multi-column and multi-row GeneratedKeys result set from Postgres.

**Comment by stevenschlansker** [comment]
Works as documented today as far as we know - let us know if there's more to consider here.
